Cython/Compiler/Nodes.py

```diff
@@ -3723,13 +3776,16 @@ def generate_tuple_and_keyword_parsing_code(self, args, success_label, code):
         if accept_kwd_args:
             kw_unpacking_condition = Naming.kwds_cname
         else:
-            kw_unpacking_condition = "%s && PyDict_Size(%s) > 0" % (
-                Naming.kwds_cname, Naming.kwds_cname)
+            kw_unpacking_condition = "%s && __Pyx_NumKwargs_%s(%s) > 0" % (
+                Naming.kwds_cname, self.signature.fastvar, Naming.kwds_cname)
+
+        if self.num_required_kw_args > 0:
+            kw_unpacking_condition = "likely(%s)" % kw_unpacking_condition
+        else:
+            kw_unpacking_condition = "unlikely(%s)" % kw_unpacking_condition
```

Comment:
I actually wonder if this is really unlikely enough to merit a branch hint. Maybe we should remove it? (Ok, that's a bit unrelated to this PR…)

Reply:
Agreed. Keeping the `likely()` in case that kwargs are required?

Reply:
Looks right now.